src/domain/account/account.repository.ts

```diff
@@ -58,10 +59,18 @@ export class AccountRepository implements IAccountRepository {
   async getAccount(args: {
     chainId: string;
     safeAddress: string;
-    signer: string;
+    signer: `0x${string}`;
+    authPayload: AuthPayload | undefined;
   }): Promise<Account> {
     const safeAddress = getAddress(args.safeAddress);
     const signer = getAddress(args.signer);
+    if (
+      !args.authPayload ||
+      args.authPayload.signer_address !== signer ||
+      args.authPayload.chain_id !== args.chainId
+    ) {
+      throw new UnauthorizedException();
+    }
```

Comment:
Nit: just as a side note, not needed in this PR at all: I suspect we will need to repeat this logic in other auth-protected functions. So it might make sense to extract this kind of checks to a common place (utils file, domain auth service, etc.). We could create functions there like `checkChainAndSigner(authPayload: AuthPayload): void` and call them from several places.

But as I said, it's just an idea for future features.

Reply:
Also, by having this logic isolated in a new class, we could move all the 401/403 status code tests to that class' tests (as it's done in this PR `email.controller.get-email.spec.ts` changes). 

I mean, if not, we would need to do multiple repeating tests for each function that we protect via authentication. I know the code currently has many similar tests per controller, but adding 5-6 test cases per each function we want to protect via authentication (which likely will increase in the future) would scale significantly the number of overall tests.

Reply:
I created a new `AuthorizationRepository` in a3dde8a179328b6eb275b6e2e37e182568f2e006. It has an `assertChainAndSigner` method according to your suggestion. However, I think we should keep the `403`/`401` tests as they are somewhat of an E2E implementation and (as it's authorisation-based) I think it's more robust. What do you think?

Reply:
Yeah, I'm fine with keeping the tests for both `AuthorizationRepository` and the controller.

Thanks for integrating the suggestion in the PR! 😄